lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -198,7 +212,22 @@ def _on_restore_action(self, event) -> None:
             event.fail(f"Cannot restore backup {pbm_status.message}.")
             return
 
-        # TODO restore backup
+        # sometimes when we are trying to restore pmb can be resyncing, so we need to retry
+        for attempt in Retrying(
+            stop=stop_after_attempt(RESTORE_MAX_ATTEMPTS),
+            wait=wait_fixed(RESTORE_ATTEMPT_COOLDOWN),
+            reraise=True,
+            before_sleep=lambda retry_state: logger.error(
+                f"Attempt {retry_state.attempt_number} failed. {RESTORE_MAX_ATTEMPTS - retry_state.attempt_number} attempts left. Retrying after {RESTORE_ATTEMPT_COOLDOWN} seconds."
+            ),
+        ):
+            with attempt:
+                restore_started, message = self._try_to_restore(backup_id, event)
```

Comment:
This seems a typo, the function now only takes one arg, backup_id, not the event anymore, right?